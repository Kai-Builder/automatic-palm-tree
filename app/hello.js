import Links from './links';
import React from 'react';

const Hello = () => (
    <div>
        <div>{'Hello, world!'}</div>
        <Links/>
    </div>
);

export default Hello;
