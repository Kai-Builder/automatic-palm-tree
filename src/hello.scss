$breakpoint-tablet: 60em;

.hello {
  border: 0.01em solid deepskyblue;

  display: grid;
  grid-template-areas:
    'message name';

    grid-template-columns: repear(2, 1fr);
}

.message {
  grid-area: message;
}

.name {
  grid-area: name;
}

@mixin media($width) {
  @media only screen and (max-width: $width) {
    @content;
  }
}

@include media($breakpoint-tablet) {
  .hello {
    grid-template-areas:
    'message'
    'name';
  }
}
