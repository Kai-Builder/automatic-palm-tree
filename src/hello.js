import {hello, message, name} from './hello.scss';
import PropTypes from 'prop-types';
import React from 'react';

export const Hello = ({className}) => (
  <h1
    className={`${className} ${hello}`}
    data-test='hello'
  >
    <div className={message}>
      {`${process.env.MESSAGE}`}
    </div>
    <div className={name}>
      {`${process.env.NAME}`}
    </div>
  </h1>
);

Hello.propTypes = {
  className: PropTypes.string
};

Hello.defaultProps = {
  className: ''
};
