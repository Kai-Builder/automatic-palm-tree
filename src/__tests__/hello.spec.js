import {Hello} from '../hello';
import React from 'react';
import {shallow} from 'enzyme';

describe('Hello', () => {
  const requiredProps = () => ({
  });
  const render = (props = requiredProps()) => shallow(<Hello {...props} />);

  test('component renders', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });
});

