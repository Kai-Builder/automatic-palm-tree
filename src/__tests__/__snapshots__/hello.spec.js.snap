// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component has a welcome message 1`] = `
<h1
  className="hello"
  data-test="hello"
>
  Hello, world!
</h1>
`;

exports[`Hello component renders 1`] = `
<h1
  className="hello"
  data-test="hello"
>
  Hello, World!
</h1>
`;
